dmutils/content_loader.py

```diff
@@ -4,49 +4,49 @@
 import os
 
 
-class ContentLoader(object):
-
-    def __init__(self, manifest, content_directory):
-
-        section_order = self._read_yaml_file(manifest)
+class ContentBuilder(object):
 
+    def __init__(self, manifest, content_directory, yaml_loader):
+        self.yaml_loader = yaml_loader
+        section_order = yaml_loader.read(manifest)
         self._directory = content_directory
-        self._question_cache = {}
-        self.sections = [
+
+        self._all_sections = [
             self._populate_section(s) for s in section_order
         ]
 
-    def get_section(self, requested_section):
+        self.sections = self._all_sections
 
-        for section in self.sections:
+    def get_section(self, requested_section):
+        for section in self._all_sections:
             if section["id"] == requested_section:
                 return section
+        return None
 
     def get_question(self, question):
+        return self.yaml_loader.read(self._directory + question + ".yml")
 
-        if question not in self._question_cache:
-
-            question_file = self._directory + question + ".yml"
+    def filter(self, service_data):
+        self.sections = [
+            section for section in [
+                self._get_section_filtered_by(section["id"], service_data)
+                for section in self._all_sections
+            ] if section is not None
+        ]
 
-            if not self._yaml_file_exists(question_file):
-                self._question_cache[question] = {}
-                return {}
+    def get_next_section_id(self, section_id):
 
-            question_content = self._read_yaml_file(question_file)
-            question_content["id"] = question
-            self._question_cache[question] = question_content
+        previous_section_is_current = False
 
-        return self._question_cache[question]
+        for section in self.sections:
+            if previous_section_is_current:
+                return section["id"]
+            if section["id"] == section_id:
+                previous_section_is_current = True
 
-    def get_sections_filtered_by(self, service_data):
-        return [
-            section for section in [
-                self.get_section_filtered_by(section["id"], service_data)
-                for section in self.sections
-            ] if section is not None
-        ]
+        return None
 
-    def get_section_filtered_by(self, section_id, service_data):
+    def _get_section_filtered_by(self, section_id, service_data):
```

Comment:
```python
 if len(filtered_questions):
             section["questions"] = filtered_questions
             return section
```

modifies the shared `section` dictionary in the `yaml_loader._cache` – the filtered questions will be missing for all future content builders that would use the same yaml loader